### PR TITLE
[ENG-804, ENG-802, ENG-800] Minor UI Fixes

### PR DESCRIPTION
**Pull request by ameer2468**
- Files now have a tooltip to show file extension
- maxLength set for creating tag names and indexer rule names
- Max width and truncation set for location names in the topbar
- Truncation for location names in locations page
- Hex code regex validation for tags
- Tag color input now updates with color picker.

Screenshot 2023-06-24 at 4 49 13 PM

Screenshot 2023-06-25 at 5 12 11 PM

Screenshot 2023-06-25 at 4 03 45 PM

Screenshot 2023-06-24 at 4 49 45 PM

Screenshot 2023-06-24 at 7 11 20 PM

Screenshot 2023-06-24 at 4 56 12 PM

### CODE REVIEW

File: interface/app/$libraryId/settings/library/tags/CreateDialog.tsx [hunk 1]
```diff
@@ -48,6 +48,7 @@ export default (props: UseDialogProps & { assignToObject?: number }) => {
 				<Input
 					{...form.register('name', { required: true })}
 					placeholder="Name"
+					maxLength={24}
```
Comment (utkubakir) [inline interface/app/$libraryId/settings/library/tags/CreateDialog.tsx:51]: Let's also put this restriction with zod too. `.max(24)` and `min(1)` should do it. Also no whitespace etc. Actually, we should remove it from here and let zod handle the validation

File: interface/app/$libraryId/settings/library/tags/EditForm.tsx [hunk 2]
```diff
@@ -9,7 +9,10 @@ import DeleteDialog from './DeleteDialog';
 
 const schema = z.object({
 	name: z.string().nullable(),
```
Comment (utkubakir) [inline interface/app/$libraryId/settings/library/tags/EditForm.tsx:11]: zod validation too

File: interface/app/$libraryId/settings/library/locations/IndexerRuleEditor/RulesForm.tsx [hunk 3]
```diff
@@ -141,6 +141,7 @@ const RulesForm = ({ onSubmitted }: Props) => {
 					form={formId}
 					size="md"
 					placeholder="Name"
+					maxLength={18}
```
Comment (utkubakir) [inline interface/app/$libraryId/settings/library/locations/IndexerRuleEditor/RulesForm.tsx:144]: let's use zod instead

File: interface/app/$libraryId/settings/library/tags/EditForm.tsx [hunk 4]
```diff
@@ -22,30 +25,33 @@ export default ({ tag, onDelete }: Props) => {
...
-
-				<Input label="Name" {...form.register('name')} />
+			<div className="flex justify-between">
+				<div className="mb-10 flex w-24 flex-row space-x-3">
+					<Input
+						label="Color"
+						className="w-28"
+						value={form.watch('color') ?? '#ffffff'}
+						icon={<ColorPicker control={form.control} name="color" />}
+						maxLength={7}
```
Comment (utkubakir) [inline interface/app/$libraryId/settings/library/tags/EditForm.tsx:49]: zod

**Review by utkubakir — CHANGES_REQUESTED** [review]

File: interface/app/$libraryId/settings/library/tags/CreateDialog.tsx (hunk 1 above)
Comment (ameer2468) [reply interface/app/$libraryId/settings/library/tags/CreateDialog.tsx:51]: Doesn't work

File: interface/app/$libraryId/Explorer/File/RenameTextBox.tsx [hunk 5]
```diff
@@ -180,7 +192,7 @@ export const RenameTextBoxBase = forwardRef<HTMLDivElement | null, Props>(
 				onKeyDown={handleKeyDown}
 				{...props}
 			>
-				{text}
+				{extension ? <Tooltip label={extension}>{text}</Tooltip> : text}
```
Comment (utkubakir) [inline interface/app/$libraryId/Explorer/File/RenameTextBox.tsx:195]: We should show the tooltip when the text gets truncated, not when the object has an extension.

Comment (ameer2468) [reply interface/app/$libraryId/Explorer/File/RenameTextBox.tsx:195]: This was done to check whether its a folder or a file, thats why I added extension as a prop. This component is being used for files and folders.

Comment (utkubakir) [reply interface/app/$libraryId/Explorer/File/RenameTextBox.tsx:195]: image

James explained perfectly here

**Review by utkubakir — DISMISSED** [review]

**Review by utkubakir — APPROVED** [review]

**Pull request merged**